### PR TITLE
Dev fix nightly

### PR DESCRIPTION
**Pull request by 0xc0170**
Some fixes, in progress as there are still some errors to be resolved and review the current fixes.

### CODE REVIEW

File: src/zinc/os/debug.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ mod internal {
   /// A token to ensure the life of the reference to the debugging output backend
   /// doesn't outlive the backend itself.
   #[must_use]
-  pub struct Token<'a> {
+  pub struct Token {
```
Comment (0xc0170) [inline src/zinc/os/debug.rs]: Not certain about this removal, was not used, so there was an error . The another option is to use phantom data feature?

File: src/zinc/hal/stm32l1/usart.rs [hunk 2]
```diff
@@ -131,16 +131,17 @@ impl CharIO for Usart {
...
-  fn write_str(&mut self, s: &str) -> fmt::Result {
-    use core::str::StrExt;
-    for b in s.bytes() {
-      wait_for!(self.reg.sr.transmit_data_empty());
-      self.reg.dr.set_data(b as u16);
-    }
-    Result::Ok(())
-  }
-}
+// TODO: encoding::types Writer
```
Comment (0xc0170) [inline src/zinc/hal/stm32l1/usart.rs]: Writer was moved to String, will need to be fixed

**Comment by 0xc0170** [comment]
Is it just me, or the warning list is so long ,that errors are not shown in Travis? Looks like 10k is a limit?  :8ball:

File: src/zinc/hal/stm32l1/usart.rs (hunk 2 above)
Comment (farcaller) [inline src/zinc/hal/stm32l1/usart.rs]: drop your nickname in TODO (`TODO(0xc0170): ...`) so the authorship is preserved please.

Comment (0xc0170) [inline src/zinc/hal/stm32l1/usart.rs]: I'll do if this one stays unfixed. That one was not supposed to be pushed though :-)

File: src/zinc/util/volatile_cell.rs [hunk 3]
```diff
@@ -22,7 +22,6 @@ use core::intrinsics::{volatile_load, volatile_store};
 /// It is mostly used by the ioreg family of macros.
 pub struct VolatileCell<T> {
   value: T,
-  invariant: InvariantType<T>,
```
Comment (farcaller) [inline src/zinc/util/volatile_cell.rs]: `InvariantType` use can go away as well.

**Comment by bgamari** [comment]
Thanks for doing this!

**Comment by 0xc0170** [comment]
@simias Can you look at Regex and its usage in your tiva port? 2 peripherals are using it, if possible to replace it with what rust currently provides?

**Comment by 0xc0170** [comment]
@bgamari You might know how to solve that regex and some other errors, right now are in platformtree. will you find time to look at this?

**Comment by simias** [comment]
I thought regexes were convenient for parsing the platform tree, is it really annoying to maintain?

Of course it could be replaced with "manual" parsing but it would make it much more tedious and potentially bogus (and harder to maintain IMO)

Unless we standardize on a well defined grammar and have helper functions to parse the entries for all platforms, that might actually help with consistency.

**Comment by farcaller** [comment]
I'm working on porting the build process over to cargo where that will be a non-issue.

**Comment by simias** [comment]
Sounds great! :)

**Pull request merged**